agent/xds/config.go

```diff
@@ -175,24 +175,40 @@ func ParseGatewayConfig(m map[string]interface{}) (GatewayConfig, error) {
 	return cfg, err
 }
 
-// Return an envoy.OutlierDetection populated by the values from this struct.
+// Return an envoy.OutlierDetection populated by the values from structs.PassiveHealthChec.
 // If all values are zero a default empty OutlierDetection will be returned to
 // enable outlier detection with default values.
-func ToOutlierDetection(p *structs.PassiveHealthCheck) *envoy_cluster_v3.OutlierDetection {
+// If override is not nil, it will overwrite the values from p, e.g., ingress gateway defaults
+func ToOutlierDetection(p *structs.PassiveHealthCheck, override *structs.PassiveHealthCheck) *envoy_cluster_v3.OutlierDetection {
 	od := &envoy_cluster_v3.OutlierDetection{}
-	if p == nil {
-		return od
+	if p != nil {
+
+		if p.Interval != 0 {
+			od.Interval = durationpb.New(p.Interval)
+		}
+		if p.MaxFailures != 0 {
+			od.Consecutive_5Xx = &wrappers.UInt32Value{Value: p.MaxFailures}
+		}
+
+		// NOTE: EnforcingConsecutive5xx must be great than 0
+		if p.EnforcingConsecutive5xx != nil && *p.EnforcingConsecutive5xx > 0 {
```

Comment:
I believe this ends up getting rid of the ability for users to set `EnforcingConsecutive5xx` to zero which is a breaking change for other proxy types.

Reply:
Added some logic to handle the legacy scenarios. Please take another look. Thanks.